src/electos/ballotmaker/data/extractor.py

```diff
@@ -0,0 +1,311 @@
+            names = []
+            parties = []
+            _party_ids = set()
+            if selection.candidate_ids:
+                for id_ in selection.candidate_ids:
+                    candidate = self._index.by_id(id_)
+                    name = self._candidate_name(candidate)
+                    if name:
+                        names.append(name)
+                    party, _party_id = self._candidate_party(candidate)
+                    if party:
+                        parties.append(party)
+                        _party_ids.add(_party_id)
+            # If there's only one party ID, all candidates share the same party.
+            # If there's any divergence track them all individually.
+            if len(_party_ids) == 1:
+                parties = parties[:1]
+            result = {
+                "id": selection.model__id,
+                "name": names,
+                "party": parties,
+                "is_write_in": bool(selection.is_write_in)
+            }
+            candidates.append(result)
+        return candidates
+
+
+    def _candidate_contest_offices(self, contest: CandidateContest):
+        """Get any offices associated with a candidate contest."""
+        offices = []
+        if contest.office_ids:
+            for id_ in contest.office_ids:
+                office = self._index.by_id(id_)
+                name = _text_content(office.name)
+                offices.append(name)
+        return offices
+
+
+    def _candidate_contest_parties(self, contest: CandidateContest):
+        """Get any parties associated with a candidate contest."""
+        parties = []
+        if contest.primary_party_ids:
+            for id_ in contest.primary_party_ids:
+                party = self._index.by_id(id_)
+                name = _text_content(party.name)
+                parties.append(name)
+        return parties
+
+
+    def _contest_election_district(self, contest: Contest):
+        """Get the district name of a contest."""
+        district = self._index.by_id(contest.election_district_id)
+        district = _text_content(district.name)
+        return district
+
+
+    def _candidate_contest_of(self, contest: CandidateContest):
+        """Extract candidate contest subset needed for a ballot."""
+        district = self._contest_election_district(contest)
+        offices = self._candidate_contest_offices(contest)
+        parties = self._candidate_contest_parties(contest)
+        candidates = self._candidate_contest_candidates(contest)
+        result = {
+            "id": contest.model__id,
+            "type": "candidate",
+            "title": contest.name,
+            "district": district,
+            "vote_type": contest.vote_variation.value,
+            # Include even when default is 1: don't require caller to track that.
+            "votes_allowed": contest.votes_allowed,
+            "candidates": candidates,
+            # "offices": offices,
+            # "parties": parties,
```

Comment:
The way we've ended up design the ballot data, we don't track parties or offices on candidate contests. (We do on the candidate contest _selections_ (aka choices)). But for a while it looked like we would, so instead of deleting what we might come back and use I commented it out. I should have commented out the calls to get them too, but I was verifying those worked, and lost track.

I'll comment out what's unused. But it's tested and I don't want to delete it completely.